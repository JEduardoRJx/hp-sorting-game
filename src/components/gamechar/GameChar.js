import React from 'react';
import './GameChar.scss'

const GameChar = ({character, isLoading, checkAnswer, message, toggleModal}) => {
  console.log(isLoading)
  if(isLoading) {
    return null;
  }

  const popupMessage = <div className={`game-modal ${toggleModal}`}>
        <h1 className='game-modal-h1'>{message}</h1>
      </div>

  return (
    <article className='game-char'>
        {popupMessage}
        <h1 className='game-character-name'>{character.name}</h1>
        <img className='game-crest' 
          name='Gryffindor' 
          src={require('../../images/gryffindor.png')} 
          onClick={event => checkAnswer(event)}/>
        <img className='game-crest' 
          name='Slytherin' 
          src={require('../../images/slytherin.png')} 
          onClick={event => checkAnswer(event)}/>
        <img className='game-crest' 
          name='Hufflepuff' 
          src={require('../../images/hufflepuff.png')} 
          onClick={event => checkAnswer(event)}/>
        <img className='game-crest' 
          name='Ravenclaw' 
          src={require('../../images/ravenclaw.png')} 
          onClick={event => checkAnswer(event)}/>
      </article>
  )
}

export default GameChar;