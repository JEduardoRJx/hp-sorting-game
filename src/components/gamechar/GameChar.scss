$hw-dark: #4c3638;
$hw-light: #D3A481;
$sl-dark: #276F45;
$sl-light: #A8B0BF;
$gryf-dark: #702737;
$gryf-light: #DAA520;
$rav-dark: #274F70;
$rav-light: #C98C4E;
$huff-dark: #141414;
$huff-light: #D1B738;

.game-char {
  grid-column: 2/4;
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr;
  justify-items: center;
  align-items: center;
}

.game-crest {
  height: 225px;
  &:hover {
    transform: scale(1.2);
    cursor: pointer;
  }
}

.game-character-name {
  font-size: 3em;
  grid-column: 1/3;
  color: $hw-light;
}

.game-modal {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: $hw-light;
  color: $hw-dark;
  width: 100px;
  font-size: 4em;
  height: 150px;
  width: 55%;
  margin: 1% 0 0% 5%;
  // border: solid 5px $hw-light;
  border-radius: 25px;
}

.hide {
  display: none;
  position: absolute
}

.show {
  z-index: 1;
  position: absolute
}