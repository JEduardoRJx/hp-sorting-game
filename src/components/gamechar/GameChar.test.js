import React from 'react';
import { shallow } from 'enzyme';
import { GameChar, mapStateToProps } from './GameChar';

describe ('GameChar', () => {

  describe('GameChar component', () => {
    let wrapper;
    const mockState = {
      state: {
        index: 0,
        message: 'message',
        toggleModal: 'show'
      }
    }
    const mockCheckAnswer = jest.fn();
    const mockUser = {
      name: 'name',
      wizardkind: 'wizard',
      wand: 'wand',
      house: 'slytherin'
    }

    beforeEach(() => {
      wrapper = shallow(<GameChar 
        character={mockState.index} 
        isLoading={true}
        checkAnswer={mockCheckAnswer} 
        message={mockState.message} 
        toggleModal={mockState.toggleModdal} 
        user={mockUser}
        />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it.skip('should call the checkAnswer prop when an image is clicked on', () => {
      wrapper.find('img').simulate('click');
      expect(checkAnswer).toHaveBeenCalled();
    })

  });

    describe('mapStateToProps', () => {
      it('should return a user object', () => {
        const mockState = {
          filter: 'SET_USER',
          user: {
            name: 'name',
            wizardkind: 'wizard',
            house: 'house',
            wand: 'wand'
          }
        }
        const expected = {
          user: {
            name: 'name',
            wizardkind: 'wizard',
            house: 'house',
            wand: 'wand'
          }
        }
        const mappedProps = mapStateToProps(mockState);
        expect(mappedProps).toEqual(expected);
      });
    })
});