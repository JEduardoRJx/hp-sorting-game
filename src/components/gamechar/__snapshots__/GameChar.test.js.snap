// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameChar GameChar component should match the snapshot 1`] = `""`;
